events/structs.go

```diff
@@ -15,61 +15,192 @@
 package events
 
 import (
-	"fmt"
-
+	"encoding/json"
 	"github.com/pkg/errors"
+	"github.com/ystia/yorc/helper/consulutil"
+	"github.com/ystia/yorc/log"
+	"path"
+	"strings"
+	"time"
 )
 
-// StatusUpdateType is the status update type
-type StatusUpdateType uint64
+//go:generate go-enum -f=structs.go --lower
+
+// StatusChangeType x ENUM(
+// Instance,
+// Deployment,
+// CustomCommand,
+// Scaling,
+// Workflow,
+// WorkflowStep
+// AlienTask
+// )
+type StatusChangeType int
+
+// Info allows to provide custom/specific additional information for event
+type Info map[InfoType]interface{}
+
+// InfoType represents Event status change information type
+type InfoType int
 
 const (
-	// InstanceStatusChangeType is the StatusUpdate type for an instance state change event
-	InstanceStatusChangeType StatusUpdateType = iota
-	// DeploymentStatusChangeType is the StatusUpdate type for an deployment status change event
-	DeploymentStatusChangeType
-	// CustomCommandStatusChangeType is the StatusUpdate type for an custom command status change event
-	CustomCommandStatusChangeType
-	// ScalingStatusChangeType is the StatusUpdate type for an scaling status change event
-	ScalingStatusChangeType
-	// WorkflowStatusChangeType is the StatusUpdate type for an workflow status change event
-	WorkflowStatusChangeType
+	// EDeploymentID is event information related to deploymentIS
+	EDeploymentID InfoType = iota
+	// EStatus is event information related to status
+	EStatus
+	// ETimestamp is event timestamp
+	ETimestamp
+	// EType is type event information
+	EType
+	// EWorkflowID is event information related to workflow
+	EWorkflowID
+	// ETaskID is event information related to task
+	ETaskID
+	// ENodeID is event information related to node
+	ENodeID
+	// EInstanceID is event information related to instance
+	EInstanceID
+	// EOperationName is event information related to operation
+	EOperationName
+	// ETargetNodeID is event information related to target node
+	ETargetNodeID
+	// ETargetInstanceID is event information related to target instance
+	ETargetInstanceID
+	// ETaskExecutionID is event information related to task execution
+	ETaskExecutionID
+	// EWorkflowStepID is event information related to workflow step
+	EWorkflowStepID
 )
 
-// StatusUpdate represents status change event
-type StatusUpdate struct {
-	Timestamp    string `json:"timestamp"`
-	Type         string `json:"type"`
-	Node         string `json:"node,omitempty"`
-	Instance     string `json:"instance,omitempty"`
-	TaskID       string `json:"task_id,omitempty"`
-	DeploymentID string `json:"deployment_id"`
-	Status       string `json:"status"`
+func (i InfoType) String() string {
+	switch i {
+	case EDeploymentID:
+		return "deploymentId"
+	case EStatus:
+		return "status"
+	case ETimestamp:
+		return "timestamp"
+	case EType:
+		return "type"
+	case EWorkflowID:
+		return "workflowId"
+	case ETaskID:
+		return "alienExecutionId"
```

Comment:
seems there is a mismatch where ETaskID returns alienExecutionId here and below ETaskExecutionID returns alienTaskId

Reply:
It's the confuse stuff about what Alien names an execution and Yorc a Task.
Same thing for Alien task and execution.

Reply:
Ah ok ! Maybe we could have a comment here to describing this naming issue, so that the reader understands its's done on purpose and it's not a bug